### PR TITLE
feat: Pandas v2 compatibility

### LINKED ISSUE #3709: Pandas 2.0 support
**Is your feature request related to a problem? Please describe.**

I'm trying to install feast with pandas 2.0.3, but feast is pinned to pandas `<2`: https://github.com/feast-dev/feast/blob/c75a01fce2d52cd18479ace748b8eb2e6c81c988/setup.py#L55

**Describe the solution you'd like**

Allow pandas >2. I'm not sure how much effort it will be, my experience upgrading has been simple, and provided there's a good test coverage of the pandas features being used by feast, it should be fine.

**Describe alternatives you've considered**

I could rollback my codebase to an older version of pandas but I'm using some of the new feature and rely on recent bug fixes.

**Additional context**

```
python3.9 -m venv --clear tmp
source tmp/bin/activate/
pip install pandas==2.0.3 feast==0.31.1

ERROR: Cannot install feast==0.31.1 and pandas==2.0.3 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested pandas==2.0.3
    feast 0.31.1 depends on pandas<2 and >=1.4.3


```

### PR DESCRIPTION
**Pull request by sudohainguyen**
**What this PR does / why we need it**:
Resume works from #3928

**Which issue(s) this PR fixes**:

Fixes #3709

### CODE REVIEW

**Comment by sudohainguyen** [comment]
pending on this [PR](https://github.com/snowflakedb/snowflake-connector-python/pull/1872) to be released from snowflake connector

File: setup.py
Comment (tokoko) [inline setup.py:1]: I think you can remove pandavro and fastavro from requirements. I couldn't find any usage in the codebase.

Comment (sudohainguyen) [reply setup.py:1]: yeah no usage in codebase but there is a hiccup in pandavro v1.7.0 we should ignore
https://github.com/ynqa/pandavro/blob/v1.7.0/setup.py

Comment (tokoko) [reply setup.py:1]: sure, but we aren't using either `pandavro` nor `fastavro` and looking at the requirements, seems like there are no transitive dependencies pointing to them as well. Wouldn't removing them be a better fix?

Comment (sudohainguyen) [reply setup.py:1]: agree that should be removed to reduce deps complexity, on it
but it's not a way to fix the PR

Comment (sudohainguyen) [reply setup.py:1]: decided to skip some tests with conditions 😄

**Comment by sudohainguyen** [comment]
> pending on this [snowflakedb/snowflake-connector-python#1872](https://github.com/snowflakedb/snowflake-connector-python/pull/1872) to be released to resolve deps conflicts
> 
> Note: test cases won't be success unless pandas 2.2.0 is installed

decided to skip some tests with conditions

**Comment by sudohainguyen** [comment]
good, all passed @tokoko

File: sdk/python/requirements/py3.8-ci-requirements.txt [hunk 1]
```diff
@@ -555,7 +551,7 @@ pandas==1.5.3
     #   great-expectations
     #   pandavro
     #   snowflake-connector-python
-pandavro==1.5.2
+pandavro==1.8.0
```
Comment (tokoko) [inline sdk/python/requirements/py3.8-ci-requirements.txt:554]: These files probably need to be regenerated. `pandavro` should not be here.

Comment (sudohainguyen) [reply sdk/python/requirements/py3.8-ci-requirements.txt]: resolved

**Comment by tokoko** [comment]
@sudohainguyen probably best to wait for snowflake release anyway, don't you think? Right now this pr would bump python to 2.2.0, but CI is testing 2.1.4 (plus with some tests being skipped). btw, we should probably consider running tests independently for each "backend" at some point in the future to avoid these scenarios.

**Comment by sudohainguyen** [comment]
think we can merge, but our release will be pending until Snowflake completes their upgrade.

mark skip should be kept so other contributors can be aware of it

**Comment by sudohainguyen** [comment]
> we should probably consider running tests independently for each "backend" at some point in the future to avoid these scenarios.

agree but a lot to do so

File: sdk/python/tests/integration/offline_store/test_feature_logging.py [hunk 2]
```diff
@@ -28,6 +28,11 @@
 @pytest.mark.integration
 @pytest.mark.universal_offline_stores
 @pytest.mark.parametrize("pass_as_path", [True, False], ids=lambda v: str(v))
+@pytest.mark.skipif(
```
Comment (tokoko) [inline sdk/python/tests/integration/offline_store/test_feature_logging.py:31]: Does 2.0.3 also contain this bug? If so, even after snowflake is released, 3.8 tests will still fail once we remove `skipifs` as the latest pandas release in 3.8 is 2.0.3. Will we need to pin pandas version below 2.0.0 only for python 3.8 using environment markers?

Comment (sudohainguyen) [reply sdk/python/tests/integration/offline_store/test_feature_logging.py:31]: well actually the bug appears running [pandas testing](https://pandas.pydata.org/docs/reference/testing.html) only,
it does not affect to features though

Comment (tokoko) [reply sdk/python/tests/integration/offline_store/test_feature_logging.py:31]: still, wdyt about adding something like this in our `ci` extras: `pandas < 2.0.0 : python_version == '3.8'`. Would be better than skipping the tests entirely.

**Comment by sudohainguyen** [comment]
ok one minute

**Review by tokoko — APPROVED** [review]

**Comment by sudohainguyen** [comment]
This is good to go @franciscojavierarceo 
Thanks boss

**Comment by jeremyary** [comment]
Thanks @sudohainguyen! If you can assist with conflicts, I'd be happy to roll it in.

**Comment by sudohainguyen** [comment]
@jeremyary  sure will do 😄 
think you can have a look at #3950 first, need your help as well

**Comment by jeremyary** [comment]
@sudohainguyen thanks! I'm working a vulnerability issue this morning, but will add 3950 to today's list for myself to TAL.

**Comment by jeremyary** [comment]
@sudohainguyen got 3950 all buttoned up yesterday. 👍  I'll keep an eye here for if/when you get a chance to look at conflicts.

**Comment by sudohainguyen** [comment]
@jeremyary all good now 😄

**Review by jeremyary — APPROVED** [review]
changes proposed were already approved prior in diff review, and now this lgtm on conflict resolution. Thanks for the contrib!

**Comment by matankley** [comment]
Hi guys, thanks for the great work. 
Do you know when is this planned to be released ? I see it was merged but no release was published yet.

**Comment by sudohainguyen** [comment]
> Hi guys, thanks for the great work. 
> 
> Do you know when is this planned to be released ? I see it was merged but no release was published yet.

@franciscojavierarceo you know this best 👀

**Comment by franciscojavierarceo** [comment]
I can release this weekend 👍

**Comment by TomSteenbergen** [comment]
@franciscojavierarceo Any news on when we can expect the new release? 🙏

**Comment by franciscojavierarceo** [comment]
@TomSteenbergen so sorry for not updating you on this sooner, I introduced a bug in the latest commit that @tokoko discovered and i'm working to fix it.

**Comment by TomSteenbergen** [comment]
I see, no worries and thanks for the update @franciscojavierarceo! Keep us posted and let me know if I can help.

**Pull request merged**